lib/core/storage/clientAdapter.ts

```diff
@@ -84,17 +99,45 @@ class ClientAdapter {
     );
   }
 
-  async createIndex(index, { indexCacheOnly = false, propagate = true } = {}) {
-    if (this.cache.hasIndex(index)) {
-      throw servicesError.get("index_already_exists", this.scope, index);
-    }
-
+  async createIndex(
+    index: string,
+    { indexCacheOnly = false, propagate = true } = {}
+  ) {
+    this.addCache(index);
     if (!indexCacheOnly) {
       await this.client.createIndex(index);
     }
+    if (propagate) {
+      global.kuzzle.emit("core:storage:index:create:after", {
+        index,
+        scope: this.scope,
+      });
+    }
+  }
 
-    this.cache.addIndex(index);
+  async createVirtualIndex(
+    index: string,
+    physicalIndex = null,
```

Comment:
You shouldn't have a default value to null if it's not allowed (see the error l124)